### PR TITLE
moved run method of assets writer main to separate files

### PR DESCRIPTION
**Pull request by schwoere**
Moved the run method of the assets_writer_main to the separate assets_writer files.
Will extract asset_writer class in the future to keep the main file small and allow re-usability and more flexibility of the asset_writer.

### CODE REVIEW

**Comment by SirVer** [comment]
@googlebot Wake up, buddy. I added martin.

File: cartographer_ros/cartographer_ros/assets_writer.cc [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+#include "tf2_ros/buffer.h"
+#include "urdf/model.h"
+
+namespace cartographer_ros {
+
+constexpr char kTfStaticTopic[] = "/tf_static";
+namespace carto = ::cartographer;
+
+template <typename T>
+std::unique_ptr<carto::io::PointsBatch> HandleMessage(
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/assets_writer.cc]: This should be put in an unnamed namespace since it is only used here.

File: cartographer_ros/cartographer_ros/assets_writer.cc [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    // We use the last transform for the origin, which is approximately correct.
+    points_batch->origin = sensor_to_map * Eigen::Vector3f::Zero();
+  }
+  if (points_batch->points.empty()) {
+    return nullptr;
+  }
+  return points_batch;
+}
+
+void RunAssetsWriterPipeline(const std::string &pose_graph_filename,
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/assets_writer.cc]: Revert the change from `Foo& foo` to `Foo &foo` here and everywhere else in the file. We should run clang-format with `--style=Google`. Maybe we can add a `.clang-format` file to make this the default for this repository. This probably should be done in a separate PR though.

File: cartographer_ros/cartographer_ros/assets_writer.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+// configures a point processing pipeline and
+// pushes the points of the scans through it
+void RunAssetsWriterPipeline(const std::string &pose_graph_filename,
+                             const std::vector<std::string> &bag_filenames,
+                             const std::string &configuration_directory,
+                             const std::string &configuration_basename,
+                             const std::string &urdf_filename,
+                             const std::string &output_file_prefix,
+                             const bool use_bag_transforms);
```
[the remark this answers is not in the document]
Comment (schwoere) [reply cartographer_ros/cartographer_ros/assets_writer.h]: Done

File: cartographer_ros/cartographer_ros/assets_writer.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <string>
+#include <vector>
+
+#ifndef CARTOGRAPHER_ROS_ASSETS_WRITER_H_
+#define CARTOGRAPHER_ROS_ASSETS_WRITER_H_
+
+namespace cartographer_ros {
+
+// configures a point processing pipeline and
```
[the remark this answers is not in the document]
Comment (schwoere) [reply cartographer_ros/cartographer_ros/assets_writer.h]: Done.

File: cartographer_ros/cartographer_ros/assets_writer.cc [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+            << "Processed " << (message.getTime() - begin_time).toSec()
+            << " of " << duration_in_seconds << " bag time seconds...";
+      }
+      bag.close();
+    }
+  } while (pipeline.back()->Flush() ==
+           carto::io::PointsProcessor::FlushResult::kRestartStream);
+}
+
+}  // namespace cartographer_ros
```
[the remark this answers is not in the document]
Comment (schwoere) [reply cartographer_ros/cartographer_ros/assets_writer.cc]: Done.

File: cartographer_ros/cartographer_ros/assets_writer.cc (hunk 1 above)
Comment (schwoere) [reply cartographer_ros/cartographer_ros/assets_writer.cc]: Done.

**Comment by schwoere** [comment]
Applied all the requested changes. Can you have a look again, @wohe ?

File: cartographer_ros/cartographer_ros/assets_writer_main.cc [hunk 6]
```diff
@@ -14,39 +14,11 @@
...
-#include "cartographer/sensor/point_cloud.h"
-#include "cartographer/sensor/range_data.h"
-#include "cartographer/transform/transform_interpolation_buffer.h"
-#include "cartographer_ros/msg_conversion.h"
-#include "cartographer_ros/ros_map_writing_points_processor.h"
+#include "cartographer_ros/assets_writer.h"
 #include "cartographer_ros/split_string.h"
-#include "cartographer_ros/time_conversion.h"
-#include "cartographer_ros/urdf_reader.h"
+
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: Style: Remove extra empty line.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.
